src/merge.jl

```diff
@@ -0,0 +1,236 @@
+using DataStructures # MutableBinaryHeap not found without this line. Why?
+
+# There's a module-level conflict with RegionTrees.vertices. 
+# Is there a better way to resolve it?
+import MetaGraphs
+vertices = MetaGraphs.vertices
+
+"""
+    seg_img = merge(seg, threshold)
+
+Merges segments in a [`SegmentedImage`](@ref) by building a region adjacency
+graph (RAG) and merging segments connected by edges with weight less than 
+`threshold`.
+
+# Arguments:
+* `seg`         : SegmentedImage to be merged.
+* `threshold`   : Upper bound of the adjacent segment color difference to 
+                  consider merging segments.
+
+# Citation:
+Vighnesh Birodkar
+"Hierarchical merging of region adjacency graphs"
+https://vcansimplify.wordpress.com/2014/08/17/hierarchical-merging-of-region-adjacency-graphs/
+"""
+function merge(seg::SegmentedImage, threshold::Number)::SegmentedImage
+    g = seg_to_graph(seg)
+
+    # Populate a heap of all the edges, and a Bool indicating whether the edge
+    # is valid. All edges are initially valid. The reason for this is that heap
+    # removal would be expensive, so instead, we invalidate the edge entry in the
+    # heap. 
+    function weight(t::Tuple{Edge{Int64}, Bool})::Real
```

Comment:
```suggestion
    function weight(t::Tuple{Edge{Int}, Bool})::Real
```
so it runs properly on 32-bit CPUs. This change should be made everywhere Int64 appears below. (I won't annotate them.)